### PR TITLE
[PREVIEW] Rdm 3138 Add more FE Tests

### PR DESCRIPTION
**Pull request by hemantt**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] tests have been conducted using multiple browsers or is not applicable for the changes

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: test/functional-tests/utils/randomUtils.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class RandomUtils {
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/functional-tests/utils/randomUtils.js]: Was copying Apache common RandomUtils class ;) perhaps something along the lines of dataGenerationUtils could be more suitable as the purpose of the class it to generate various formats of input data

File: test/functional-tests/pageObjects/ccd-components/fields/yesNoField.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+let RandomUtils = require('../../../utils/randomUtils.js');
+
+/**
+ * TextAreaField field component deals with interactions with the CCD Yes No Field component.
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/functional-tests/pageObjects/ccd-components/fields/yesNoField.js]: fixed

File: test/functional-tests/pageObjects/ccd-components/fields/textAreaField.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+let TextField = require('../../webdriver-components/textField.js');
+let RandomUtils = require('../../../utils/randomUtils.js');
+
+/**
+ * TextAreaField field component deals with interactions with the CCD Text Area Field component.
+ *
+ * returning this object will give access to the expected 'show your answers' page answer for the specified field
+ */
+class TextAreaField{
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/functional-tests/pageObjects/ccd-components/fields/textAreaField.js]: fixed

File: test/functional-tests/pageObjects/ccd-components/fields/fixedList.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+
+let Dropdown = require('../../webdriver-components/dropdown.js');
+
+/**
+ * CCD Fixed List dropdown field component
+ */
+class FixedList {
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/functional-tests/pageObjects/ccd-components/fields/fixedList.js]: fixed

File: test/functional-tests/pageObjects/ccd-components/fields/dateField.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
+let RandomUtils = require('../../../utils/randomUtils.js');
+TextField = require('../../webdriver-components/textField.js');
+
+
+/**
+ * CCD date field component, encapsulated all interactions with the Date field
+ */
+class DateField{
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/functional-tests/pageObjects/ccd-components/fields/dateField.js]: fixed

File: test/functional-tests/config/local.conf.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  frameworkPath: require.resolve('protractor-cucumber-framework'),
+
+  capabilities: {
+    'shardTestFiles': true,
+    'maxInstances': 1,
+    'acceptInsecureCerts': true,
+    'marionette': false,
+    'browserName': 'chrome',
+
+    // chromeOptions: { args: ['--headless', '--no-sandbox', '--disable-dev-shm-usage', '--disable-setuid-sandbox', '--no-zygote '] },
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/functional-tests/config/local.conf.js]: removed

File: test/functional-tests/config/local.conf.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+    // 'proxy': {
+    //   'proxyType': 'manual',
```
[the remark this answers is not in the document]
Comment (noronhaa) [reply test/functional-tests/config/local.conf.js]: I'll change it to uncommented, people can comment locally if testing against local docker

File: test/functional-tests/utils/randomUtils.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (venumadhavancha) [reply test/functional-tests/utils/randomUtils.js]: yeah, dataGenerationUtils may suite. Only thing in future if we add anything other than data gen, again name conflicts. Option is yours, choose something other than randomutils.

File: test/functional-tests/pageObjects/basePage.js [hunk 8]
```diff
@@ -9,88 +9,32 @@ class BasePage {
...
         }
+
+        this._formFields = 'ccd-case-edit-form > div > * > *';
+        this._optionalClass = 'ng-valid';
+        this._mandetoryClass = 'ng-invalid';
     }
 
     async waitForElementToBeVisible(element){
-        await browser.wait(EC.visibilityOf(element), TIMEOUT);
+        await browser.wait(await EC.visibilityOf(element), TIMEOUT);
```
Comment (hemantt) [inline test/functional-tests/pageObjects/basePage.js]: Not major, but can we make it much graceful with a try, catch. Rest looks good, will not hold off this PR anyway

**Pull request merged**